hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/CommonUtils.java

```diff
@@ -88,4 +135,143 @@ public static boolean productWouldOverflow(final long multiplier, final long mul
     static void setSha384HashTag(final String sha384HashTag) {
         CommonUtils.sha384HashTag = sha384HashTag;
     }
+
+    public static HederaFunctionality functionOf(final TransactionBody txn) throws UnknownHederaFunctionality {
```

Comment:
I like this. But, I believe this function exists in `MiscUtils`. You can delete there and move it into this class as you did.

Reply:
I updated the change you asked. Please re-check. Thanks.